### DISCUSSION TITLE
Welcome to ESP32-CAM-Video-Telegram Discussions!

### DISCUSSION THREAD

**Discussion by jameszah** [jameszah/ESP32-CAM-Video-Telegram#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by jameszah** [comment]
Hey, is this discussion feature new?

**Comment by dmivla** [comment]
hello!
![image](https://user-images.githubusercontent.com/91685394/135488168-1ae9c023-5f54-4774-9656-f326574c9c6a.png)


Rebooting...
ets Jun  8 2016 00:22:57

rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:1
load:0x3fff0018,len:4
load:0x3fff001c,len:1044
load:0x40078000,len:10124
load:0x40080400,len:5856
entry 0x400806a8
---------------------------------
ESP32-CAM Video-Telegram pir-cam 8.8
---------------------------------
psram_avi allocation failed
psram_idx allocation failed
Connecting Wifi: dlink
......mDNS responder started 'deskpir'

WiFi connected
IP address: 192.168.0.107
Setup PIRpin = 0, 0, 0, 0, 0,  

Got a message /clip

Start the avi ... at 11041
Framesize 8, quality 10, length 0 seconds

Starting an avi 
Guru Meditation Error: Core  1 panic'ed (StoreProhibited). Exception was unhandled.
Core 1 register dump:
PC      : 0x4000c2e4  PS      : 0x00060c30  A0      : 0x800d263c  A1      : 0x3ffdba30  
A2      : 0x00000000  A3      : 0x3ffbdbb8  A4      : 0x000000f0  A5      : 0x00000000  
A6      : 0x46464952  A7      : 0x000e01d8  A8      : 0x00000000  A9      : 0xffffffdb  
A10     : 0x00000002  A11     : 0x3ffc29b5  A12     : 0x00000000  A13     : 0x3ffd8d20  
A14     : 0x3ffc29ac  A15     : 0x00000208  SAR     : 0x0000001e  EXCCAUSE: 0x0000001d  
EXCVADDR: 0x00000000  LBEG    : 0x4000c2e0  LEND    : 0x4000c2f6  LCOUNT  : 0x0000000e  

ELF file SHA256: 0000000000000000

Backtrace: 0x4000c2e4:0x3ffdba30 0x400d2639:0x3ffdba40 0x4008bdde:0x3ffdba90

Rebooting...

**Comment by jameszah** [comment]
What esp32 module are you using?  You need a module with 4MB of psram, such as a esp32 wrover module on a esp32-cam board.

Those "psram_avi allocation failed" errors suggest it could not find the psram.  You could add the code below before and after the ps_malloc statement, to see how much psram the esp32 has available.


https://github.com/jameszah/ESP32-CAM-Video-Telegram/blob/74088e263956f845a32ab7be890538385d746c1e/ESP32-CAM-Video-Telegram_8.8x.ino#L1032

```
Serial.printf("Internal Total heap %d, internal Free Heap %d\n", ESP.getHeapSize(), ESP.getFreeHeap());
Serial.printf("SPIRam Total heap   %d, SPIRam Free Heap   %d\n", ESP.getPsramSize(), ESP.getFreePsram());
```
Another possibility is you have the right hardware, but the wrong board specified.  It has to be wrover module or Ai-Thinker ESP32-CAM to make sure the psram is enabled.

![image](https://user-images.githubusercontent.com/36938190/135728179-b9450c39-fd73-442f-950c-f2c0e84b7533.png)

**Comment by TheCDB** [comment]
I have the same problem and worse telegram bot keeps sending the /clip command.

As far as I know I have the AI thinker board, added your code.
21:14:20.427 -> ESP32-CAM Video-Telegram pir-cam 8.9
21:14:20.427 -> ---------------------------------
21:14:20.427 -> Internal Total heap 301240, internal Free Heap 273164
21:14:20.427 -> SPIRam Total heap   2097100, SPIRam Free Heap   2097100
21:14:20.427 -> psram_avi allocation failed
21:14:21.038 -> Pic, len=22473, new fb 3F81D6B8
21:14:21.106 -> Pic, len=22536, new fb 3F80EF68
21:14:21.140 -> Pic, len=22371, new fb 3F800818
21:14:21.208 -> Pic, len=22433, new fb 3FFE4374
21:14:21.242 -> Pic, len=22435, new fb 3F80EF68
21:14:21.310 -> Pic, len=22532, new fb 3F800818
21:14:21.344 -> Pic, len=22337, new fb 3FFE4374
21:14:21.411 -> Connecting Wifi: Secured-V
21:14:21.581 -> .....mDNS responder started 'deskpir'
21:14:24.088 -> 
21:14:24.088 -> WiFi connected
21:14:24.088 -> IP address: 192.168.2.140
21:14:24.088 -> Setup PIRpin = 0, 0, 0, 0, 0,  
21:14:27.514 -> 
21:14:27.514 -> Got a message /clip
21:14:27.752 -> 
21:14:27.752 -> Start the avi ... at 7797
21:14:27.752 -> Framesize 8, quality 10, length 0 seconds
21:14:27.752 -> 
21:14:27.786 -> Starting an avi 
21:14:27.786 -> Guru Meditation Error: Core  1 panic'ed (StoreProhibited). Exception was unhandled.
21:14:27.820 -> Core 1 register dump:
21:14:27.820 -> PC      : 0x4008d484  PS      : 0x00060430  A0      : 0x800d25dc  A1      : 0x3ffdbf50  
21:14:27.820 -> A2      : 0x00000000  A3      : 0x3ffbdbb8  A4      : 0x000000f0  A5      : 0x00000000  
21:14:27.820 -> A6      : 0x46464952  A7      : 0x000e01d8  A8      : 0x00000000  A9      : 0x3ffdbe40  
21:14:27.820 -> A10     : 0x00000002  A11     : 0x3ffc3005  A12     : 0x3ffc1841  A13     : 0x00000000  
21:14:27.854 -> A14     : 0x0000000a  A15     : 0x3ffc2ffc  SAR     : 0x00000004  EXCCAUSE: 0x0000001d  
21:14:27.854 -> EXCVADDR: 0x00000000  LBEG    : 0x4008d480  LEND    : 0x4008d49c  LCOUNT  : 0x0000000e  
21:14:27.854 -> 
21:14:27.854 -> ELF file SHA256: 0000000000000000
21:14:27.854 -> 
21:14:27.854 -> Backtrace: 0x4008d484:0x3ffdbf50 0x400d25d9:0x3ffdbf60 0x400d2a97:0x3ffdbf80 0x4009185a:0x3ffdbfc0
21:14:27.887 -> 
21:14:27.887 -> Rebooting...
21:14:27.887 -> ets Jul 29 2019 12:21:46
21:14:27.887 -> 
21:14:27.887 -> rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
21:14:27.887 -> configsip: 0, SPIWP:0xee
21:14:27.887 -> clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
21:14:27.887 -> mode:DIO, clock div:1
21:14:27.887 -> load:0x3fff0018,len:4
21:14:27.887 -> load:0x3fff001c,len:1216
21:14:27.887 -> ho 0 tail 12 room 4
21:14:27.887 -> load:0x40078000,len:10944
21:14:27.887 -> load:0x40080400,len:6388
21:14:27.887 -> entry 0x400806b4

Regards,
Robert

**Comment by jameszah** [comment]
![image](https://user-images.githubusercontent.com/36938190/159147694-a8a3da23-eeef-49d3-bd1c-e28d99fc8b96.png)

This was set up for the 4MB of psram or spiram and you have 2MB.  It uses the spiram to store the movie before sending it to telegram.

Telegram also keeps sending the command until you properly receive it, and I think you device reboots before it can acknowledge it.

**Reply by TheCDB** [reply]
Thanks. Seems someone sold me a crap espcam.

**Reply by TheCDB** [reply]
Anyone got this to work on an ai thinner esp32 cam board?
Cause I think 2Mb is not a normal size for ps ram.

**Reply by jameszah** [reply]
You could try this:
https://github.com/jameszah/ESP32-CAM-Video-Telegram/blob/1b059b59dd29961eb4e3e1173db8489a2ce26b5e/ESP32-CAM-Video-Telegram_8.9x.ino#L1064

Change the 3000 to 1000.  It should fit within the 2MB.  The movie lengths will be shorter than advertised, but I think the rest should work.  I have not tried it yet.  The line below should end the recording before you run out of recording space.

https://github.com/jameszah/ESP32-CAM-Video-Telegram/blob/1b059b59dd29961eb4e3e1173db8489a2ce26b5e/ESP32-CAM-Video-Telegram_8.9x.ino#L740

Let us know if it works.  Someone else had the same problem a month or two ago.

**Comment by TheCDB** [comment]
Hi,
I tried XGA and that works, but if I choose a higher reolution it crashes on the frame buffer? Asking this because the comment line invludes UXGA.

What is the maximum rsolution and how to reach that? that is with 4mb psram.

**Comment by jameszah** [comment]
You can set it to uxga, but the camera driver would need more memory, so that 3000 number above would need to shrink.
You could start the camera first, and then use the leftover psram with the esp.getfreepsram to use the leftovers.  I think the frame buffer count could also be modified for the camera driver changes from esp-arduino 2.0 from a few months ago.

I choose vga as you would get 150 frames so you would have motion in the video.  I larger frame would have better detail, but if you only have 50 frames, you would get a short jerky video.

I intend on looking into that a little more as I include the changes in the driver, so I could look at that more at that time.

https://github.com/espressif/esp32-camera/issues/357#issuecomment-1047086477

//Serial.printf("SPIRam Total heap   %d, SPIRam Free Heap   %d\n", ESP.getPsramSize(), ESP.getFreePsram());

https://github.com/jameszah/ESP32-CAM-Video-Telegram/blob/1b059b59dd29961eb4e3e1173db8489a2ce26b5e/ESP32-CAM-Video-Telegram_8.9x.ino#L193

**Comment by J0E45** [comment]
hello, I'm using your program for my tool and I'm having problems with the verification and uploading process, when I try to verify, there is a notification like the following, why does this happen, please help me to solve it
![notifikasi error](https://user-images.githubusercontent.com/101226069/163703332-3f8fb3ce-87c4-40e7-8c97-929aad295afd.jpg)

**Reply by jameszah** [reply]
You have to drag these files into your arduino ide so the main program can access them -- they are there on the github in the same place as the .ino program.

That should do it, but you might have to remove the Brian Lough universal telegram bot library if you have it installed -- these are a modified version of his library.

![image](https://user-images.githubusercontent.com/36938190/163753782-6dc4cda9-74d7-44c6-95f5-a8b7f79cddfe.png)

**Reply by J0E45** [reply]
ahhh I see! yesterday I tried to find the solution myself by trying to replace the file in the src folder in the UniversalTelegrambot library from brian lough and it worked fine, thanks for the solution!

actually i want to use your program for my task but i want to replace the pear sensor with another component, i don't fully understand your program because i'm not competent in programming, i want to change some lines that you made to suit my task. What code on the line can I change?

**Comment by wafiq16** [comment]
hello, what about the psramsize said 0?? 

thanks,
wafiq

**Reply by jameszah** [reply]
if your esp32 module has no psram - this will not work.
The video is recorded in psram, and then sent to telegram.

**Reply by wafiq16** [reply]
the problem is for the first until maybe the fifth try, the esp cam doing well. but after that, psram said 0. is broken hardware caused it?

**Comment by popeyaho001** [comment]
hi, is this code need PIR module for /enpir function, I installed it, worked very good, but pir not working, sorry for my n00b

**Reply by jameszah** [reply]
Is your pir working?  Wire up an led or volt-meter to the output of the pir and make sure it is turning off and on correctly.  Then connect it to gpio 13 through a resistor.

**Reply by popeyaho001** [reply]
seem I missed a resistor, btw can you add this function https://github.com/eloquentarduino/EloquentSurveillance

**Reply by FBMinis** [reply]
> seem I missed a resistor, btw can you add this function https://github.com/eloquentarduino/EloquentSurveillance

I hope not because  it's only useful indoors, as it evaluates movement based on changes in pixels.

**Comment by kenpachi82** [comment]
thank you very much for this project which is by far the best using telegram with ESP32 cam.
I wonder if there is a way to use a 433 or 315 MHz RF module instead of the PIR module. I'm working on a project to take pictures of animals but I want to start taking pictures by sending a command by RF.

**Comment by Sudoweeb** [comment]
Hi everyone,

I am working on a project that I am completely blank on how to approach. I would appreciate any direction you can provide.

I need to use an ESP32-CAM to capture a video feed and apply scene detection and face detection in real-time. I would like to avoid using paid cloud services. I am not sure if the ESP32 could handle such computation. Is it possible to send the feed to an Android phone for processing? Or is there another way to make this device mobile?

Thanks in advance for your help and apologies if this is the wrong forum.

**Reply by jameszah** [reply]
The youtube search "face detection esp32-cam" will give you a load of videos on face detection.  The standard esp32-cam example program from espressif which lets you adjust all the camera parameters, also has a face detect algorithm in it.

You can also use the esp32-cam just to take still pictures or video and deliver that to a smaller cheaper device than an android phone -- like a raspberry pi 3 or better -- and do the face object detection there.  

This program https://github.com/jameszah/ESP32-CAM-VideoCam has some info on setting up the video or still camera and sending the data to motioneye on a raspberry pi.  

Or a youtube search for "raspberry pi object detection" will give you a heap of videos on detecting animals, cars, people, etc with Tensor Flow and various pi software, and there is capability with azure, aws, google to do much better processing in the free or nearly free category.

**Comment by Hedi2525** [comment]
Hi. This is the best ESP32cam project. For me it is running without any issue.
Have used the installer and everything is ok.
Out of curiosity what is the purpose of the 2 ino.img files and the manifest.json one inside the installer folder.
Thanks.

**Reply by jameszah** [reply]
the ".img" files are the machine code that is stored on the flash of the esp32.
(There is a old version 9.5 still sitting there.)
The manifest is the instructions to the web installer system,
You can read more about the web installer system here: 
https://github.com/witnessmenow/ESP-Web-Tools-Tutorial

**Comment by Hedi2525** [comment]
Got it. Thanks a lot.

**Reply by Hedi2525** [reply]
Hi. How to modify the latest version (version 9.6) in order to compile it using Arduino IDE (ie without using the installer way).
I'm getting the message "Error compiling for board AIThinker..)
NB: The old versions (8.8 and 8.9) are compiling and uploading without any issue
thanks.

**Reply by FBMinis** [reply]
It compiled without modification for me. Make sure you have "Show verbose output during compilation"option enabled in File -> Preferences to have more info about that error.

**Reply by Hedi2525** [reply]
very strange issue. have rechecked  and the old versions (8.8 and8.9) are compiling and uploading wthout any problem BUT the new ones (9.5 and 9.6) are giving this message : Exit status 1 . error compiling for board AIThinker ESP32cam. have tested with the wrover module board with the same result.???
Yes Show verbose output during compilation option is enabled .
As for Libraries, the only difference between the new and the old sketchs is the wifi manager library. May this be the problem?
I'm using the last revision ofthe ESP32 board.

**Reply by jameszah** [reply]
I'd guess it is something to do with folders directories.   Put all rhe files in a folder, exit and restart the ide, and open from that new folder.  And if you copied it off github, make sure it's not write protected or something strange.

**Reply by Hedi2525** [reply]
I'm just opening the .ino file with wordpad and after that copy and past the entire text (the program) into a blank page of arduino ide and that is exactly what i'm doing with the old versions.

**Reply by jameszah** [reply]
![image](https://github.com/jameszah/ESP32-CAM-Video-Telegram/assets/36938190/b3d84bcc-5baa-4a3a-bd5d-c3325ca534ed)

You need the 3 files in a folder with this 9.6 name, and saved, then open the ino

**Reply by Hedi2525** [reply]
SOLVED....
The problem was the WifiManager library. The one installed is very outdated . So i updated it to the latest version and now  it is compiling without any issue.
And yes have put the 3 files in the same folder (it didn't compile before the wifimanager library update.)
Have detected the issue by reading carefully the Arduino verbose output during compilation like you advised.
Thanks a lot for your support.

**Reply by Hedi2525** [reply]
Another question please. Why the system isn't saving the WiFi credentials to the eeprom? It switch to the AP mode after turning off and on the esp 32. The Chat Id and the Bot Token are saved but not the wifi credentials? So I have to re enter my WiFi credentials every time I turn the power off and on.
Thanks.

**Reply by jameszah** [reply]
The ssid and password are stored by the wifimanager system.  If it cannot connect on the previous parameters, it will go to AP mode to enter new parameters.  There are some settings just before wufimanager starts to try harder and wait longer to connect to your router.  You might be in a situation of tenuous signal strength - so just better signal strength will solve the problem -- such as move wires away from esp32 antenna or add an external esp32 antenna.  Or just skip the wifimanger and hardcode your ssid and password.

**Reply by Hedi2525** [reply]
Thanks for answering.
Skipping the wifi manager and hardcoding my wifi credentials means just writing  my ssid and password in the sketch itself (the .ino file) before uploading it without other modification to the sketch. Am i right ?
Sorry i'm noob..

**Reply by jameszah** [reply]
yep

**Reply by FBMinis** [reply]
I have a camera outside that is always holding on to dear life in terms of rssi when the weather is really overcast or it's raining. Since you coded it to reboot when it checks for new messages but it's not connected,, sometimes it reboots and hangs on WiFimanager. It stays like that until the weather clears up but it always reconnects, even after a couple of days.

**Comment by FBMinis** [comment]
> Thanks for answering. Skipping the wifi manager and hardcoding my wifi credentials means just writing my ssid and password in the sketch itself (the .ino file) before uploading it without other modification to the sketch. Am i right ? Sorry i'm noob..


I have a variant of v9.5 without wifimanager if you need.

**Reply by Hedi2525** [reply]
Thanks for that. Yes I do need it.. could you send it to me. Thanks again.

**Reply by FBMinis** [reply]
[ESP32-CAM-Video-Telegram_9-5_NoWM.txt](https://github.com/jameszah/ESP32-CAM-Video-Telegram/files/12716242/ESP32-CAM-Video-Telegram_9-5_NoWM.txt)

**Reply by Hedi2525** [reply]
Thanks a lot

**Reply by Hedi2525** [reply]
OK. Now that everything is working without any issue, i have installed a second ESP32Cam in another room. I have also created a second Bot using the same account (so the same Chat_Id) but of course with its own Bot Token (a second Bot Token) and programmed the second ESP32 Cam using the new Bot Token (and the same Chat_Id). The problem is that this second ESP32 Cam cannot interact with this second Bot (it is programmed with). Is there a workarround to get 2 ESP32  interacting with telegram so i can supervise what is happening in two rooms?

**Reply by Hedi2525** [reply]
Got it...
I had to create a new account.
Thanks again.

**Reply by FBMinis** [reply]
You can join all bots in the same Channel and they will be able to interact. For example, I had a PIR bot and a vibration bot in the same Channel as three cameras which would capture video+photo simultaneous when an alert message would be posted by the sensors.

You can also use IFTTT or Macrodroid app to interact with your Telegram bots, for example to get email notifications, interact with Twitter, save photos to Google Drive, etc.

**Reply by Hedi2525** [reply]
Strange but on my side the 2 PIR bots with the same chat_id and different tokens (, so using the same channel) didn't interact. So i had to create a new channel to get both of them working.

**Comment by Hedi2525** [comment]
You are right.I have now 3 PIR bots running on the same channel.
Thanks a lot for helping.

BTW, Is there the same kind of project (Motion detection associated with telegram) using Raspberry?

**Reply by FBMinis** [reply]
Cool, glad it helped. You can also add IFTTT to the channel as a bot and have it post a message, which all the cambots will see and react to. For example, IFTTT knows what time is the sun rise or sunset and sends /vslow followed by /clip and all the camera bots will capture a timelapse. It's an interesting app, maybe it can be useful to you.

**Comment by maxkcy** [comment]
Hello, I can't find a download link to the installer.

**Reply by jameszah** [reply]
You need a compatible browser - like Chrome - and click that Connect button.
It installs from the web -- you do not download an installer file.

![image](https://github.com/jameszah/ESP32-CAM-Video-Telegram/assets/36938190/05820f9e-9d9b-42df-b2d0-761815c6fb84)

**Reply by tarikjmr** [reply]
Hi!
I am facing Failed to fetch while useing One CLick Installer.
Also getting- Compilation error: fatal error: esp_eth_driver.h: No such file or directory Error while Compiling.

Arduino IDE2.2.1
Arduino ESP32 Boards- 2.0.13
esp32 - 2.0.14

Can you please help.

Thanks

**Comment by jameszah** [comment]
I don't know the "failed to fetch" issue -- i would guess some kind of virus checker or firewall kind of thing.  You could post a screen shot of that.

"esp_eth_driver" or WifiClientSecure setinscecure -- the setinsecure issue seems like an old driver.  They had it for 8266, but not esp32, and then they added it to esp32.  The "eth" stuff is entirely new to me.  A screen shot or some error messages with all the details, always helps.

**Comment by tarikjmr** [comment]
Thank you for your reply. 
After so many tries, I have found some issue with my Arduino IDE, So I have tried with Cloud version of Arduino and succesfully uploaded sketch into my ESP32 CAM. I have given all the WIFI ID & Password , It works. But after restart, it disconnect automatically and again asking to setup the WiFi. Is there any way to set the WiFi permanantly. 

ets Jul 29 2019 12:21:46

rst:0x1 (POWERON_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:1
load:0x3fff0030,len:1344
load:0x40078000,len:13864
load:0x40080400,len:3608
entry 0x400805f0
---------------------------------
ESP32-CAM Video-Telegram pir-cam 9.6
---------------------------------
PSRAM size to store the video 2065987
Good settings in the EPROM 
TelegramVDO
5871600000:AASDFhsS2eZnNMmKg13OAoJEdN_TE_P9J5D
52222222222
GMT
10
0
*wm:AutoConnect 
*wm:No wifi saved, skipping 
*wm:AutoConnect: FAILED for  103 ms
*wm:StartAP with SSID:  TelegramVDO
*wm:AP IP address: 192.168.4.1
*wm:Starting Web Portal 
*wm:config portal has timed out 
*wm:config portal exiting 
Internet failed using WiFiManager - not starting Web services
mDNS responder started 'TelegramVDO'

WiFi connected
IP address: 0.0.0.0
Camera to VGA 
try to allocate 1582987
Pic, len=36651, new fb 3F993CD0
Pic, len=12742, new fb 3F9A2CF0
Pic, len=12737, new fb 3F9B1D10
Pic, len=22955, new fb 3F9C0D30
Pic, len=20585, new fb 3F993CD0
Pic, len=20679, new fb 3F9A2CF0
Pic, len=20724, new fb 3F9B1D10
Setup PIRpin = 0, 0, 0, 0, 0,  
Sending a message to: 52222222222 at 5871600000:AASDFhsS2eZnNMmKg13OAoJEdN_TE_P9J5D
Reboot
Device: TelegramVDO
Ver: pir-cam 9.6
Rssi: 0
ip: 0.0.0.0
/start
Initial send failed, reseting WiFi Parameters
*wm:resetSettings 
*wm:SETTINGS ERASED 
***** WiFi reconnect *****
***** WiFi rerestart *****
Good settings in the EPROM 
TelegramVDO
5871600000:AASDFhsS2eZnNMmKg13OAoJEdN_TE_P9J5D
52222222222
GMT
10
0
*wm:AutoConnect 
*wm:No wifi saved, skipping 
*wm:AutoConnect: FAILED for  18 ms
*wm:StartAP with SSID:  TelegramVDO
*wm:AP IP address: 192.168.4.1
*wm:Starting Web Portal

**Comment by Josiahmphago** [comment]
hey ,i need help please.
i have loaded the code as it is ,i managed to connect to Wi-Fi manager then put in my Wi-Fi name and password, but it only connects for 5 seconds or less then it just disconnects.

the second issue is that I've put my bot id/credentials, but it does not even connect with telegram.

please your help is appreciated i enjoy this challenge just learning all this at 52yrs just fun.

**Reply by jameszah** [reply]
The WiFiManager just connects long enough to get your wifi name/pass, and telegram, and then disconnects and connects to your wifi router.  So you can check there if it connects to the router.

On the telegram, you should see messages on the Arduino serial monitor taking you through all the events of starting camera, starting wifi, connecting to telegram, getting commands from your telegram account, ....

You could cut and paste that in here, and I'll point out the errors.  You should erase your wifi and telegram name and password before pasting it here.

**Reply by Josiahmphago** [reply]
GMT
30
0
*wm:resetSettings 
*wm:SETTINGS ERASED 
*wm:AutoConnect 
*wm:No wifi saved, skipping 
*wm:AutoConnect: FAILED for  18 ms
*wm:StartAP with SSID:  AutoConnectAP
*wm:AP IP address: 192.168.4.1
*wm:Starting Web Portal

**Reply by Josiahmphago** [reply]
thank you for a quick response, may I also highlight that SSID and Password fail to save. It only saved Telegram ID and Telegram Pass.

**Reply by Josiahmphago** [reply]
ets Jul 29 2019 12:21:46

rst:0x1 (POWERON_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:1
load:0x3fff0030,len:1344
load:0x40078000,len:13964
load:0x40080400,len:3600
entry 0x400805f0

**Reply by jameszah** [reply]
I mean something like this:
Here is a wrongly typed wifi password, no ssid chosen, then using the default telegram name, which is not valid, so wifi parameters and reset and it goes back to the deskpir and 192.168.4.1 setup screen.

https://gist.github.com/jameszah/0dfab3d014c2c8d0c892e1add638f31e

**Reply by Josiahmphago** [reply]
hi, hope you good, I managed to find the issue with my telegram bot. The id had an extra number(typo).

I now have this below, but I notice you responded to someone with the same problem. I'm going to try the solution you suggested, since I'm not getting any pictures and video streaming.

id like to ask this one question, if I want to change the ssid and credentials for telegram do I still use Wi-Fi manager 162.168.4.1?  I saw another Ip address after downloading. 

thank you

/start
/flash
/flash
/photo
/photo
Reboot
Device: deskpir
Ver: pir-cam 9.6
Rssi: -53
ip: 192.168.43.177
/start
PIR Event!
Reboot
Device: deskpir
Ver: pir-cam 9.6
Rssi: -52
ip: 192.168.43.177
/start
Reboot
Device: deskpir
Ver: pir-cam 9.6
Rssi: -42
ip: 192.168.43.177
/start
Reboot
Device: deskpir
Ver: pir-cam 9.6
Rssi: -39
ip: 192.168.43.177
/start
Reboot
Device: deskpir
Ver: pir-cam 9.6
Rssi: -41
ip: 192.168.43.177
/start
Reboot
Device: deskpir
Ver: pir-cam 9.6
Rssi: -37
ip: 192.168.43.177
/start
Reboot
Device: deskpir
Ver: pir-cam 9.6
Rssi: -70
ip: 192.168.43.177
/start
Reboot
Device: deskpir
Ver: pir-cam 9.6
Rssi: -53
ip: 192.168.43.177
/start
Reboot
Device: deskpir
Ver: pir-cam 9.6
Rssi: -57
ip: 192.168.43.177
/start
Reboot
Device: deskpir
Ver: pir-cam 9.6
Rssi: -62
ip: 192.168.43.177
/start
PIR Event!
Reboot
Device: deskpir
Ver: pir-cam 9.6
Rssi: -42
ip: 192.168.43.177
/start

**Comment by Josiahmphago** [comment]
hi, can you create an app for this ESP-cam that one can use without using Telegram ?

**Reply by jameszah** [reply]
What would that app do? 🤔 
Take a picture and ....?

**Reply by Josiahmphago** [reply]
Hi James, I'd like the camera to identify pets, people, vehicles and dangerous objects like guns as an example, then notify me on the app, I should be able to add/delete users, and give access permission

I'd like the camera to have movement mechanism that I can use on the app to control the angle view.

**Reply by FBMinis** [reply]
https://dronebotworkshop.com/esp32-object-detect/

**Comment by CHE77** [comment]
Great project! 👍🏻
When will we have the pleasure of seeing the source of the latest firmware?
